checks/pinned_dependencies.go

```diff
@@ -161,6 +168,38 @@ func addPinnedResult(r *pinnedResult, to bool) {
 	}
 }
 
+func dataAsWorkflowResultPointer(data FileCbData) *worklowPinningResult {
+	pdata, ok := data.(*worklowPinningResult)
+	if !ok {
+		// panic if it is not correct type
+		panic("type need to be of worklowPinningResult")
+	}
+	return pdata
+}
+
+func createReturnValuesForGitHubActionsWorkflowPinned(r worklowPinningResult, infoMsg string,
+	dl checker.DetailLogger, err error) (int, error) {
+	if err != nil {
+		return checker.InconclusiveResultScore, err
+	}
+
+	score := checker.MinResultScore
+
+	if r.gitHubOwned != notPinned {
+		score += 2
+	}
+
+	if r.thirdParties != notPinned {
+		score += 8
+	}
+
+	if r.gitHubOwned == pinned {
```

Comment:
move the `Info()` inside the `if r.gitHubOwned != notPinned`. 
Also add a call to `Info()` inside the `if r.thirdParties != notPinned`.

Use a different message for the call to `Call()`, and use `Info3()` as in https://github.com/ossf/scorecard/blob/main/checks/permissions.go#L66. Say
```
Info3(&checker.LogMessage{
	Path: PATH2FILE, Type: checker.FileTypeSource, Offset: LINE_NO, Snippet: SNIPPET,
	Text: fmt.Sprintf("%s %s"), "GitHub-owned", infoMsg,
	})
Info3(&checker.LogMessage{
	Path: PATH2FILE, Type: checker.FileTypeSource, Offset: LINE_NO, Snippet: SNIPPET,
	Text: fmt.Sprintf("%s %s"), "Third-party", infoMsg,
	})
```

If you're not sure how to get the line numbers and snippet, omit them and just add a `//TODO: set Snippet and line numbers.`

Last point: call `createReturnValuesForGitHubActionsWorkflowPinned()` with a generic message `"actions are pinned"`